controllers/udproute_controller.go

```diff
@@ -0,0 +1,99 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/kong/blixt/pkg/vars"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	gatewayv1alpha2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
+	gatewayv1beta1 "sigs.k8s.io/gateway-api/apis/v1beta1"
+)
+
+// UDPRouteReconciler reconciles a UDPRoute object
+type UDPRouteReconciler struct {
+	client.Client
+	Scheme *runtime.Scheme
+}
+
+// Verify respective gateway has atleast one listener matching to what is specified in UDPRoute parentRefs.
+func (r *UDPRouteReconciler) verifyListener(ctx context.Context, gw *gatewayv1beta1.Gateway, udprouteSpec gatewayv1alpha2.ParentReference) error {
+	for _, listner := range gw.Spec.Listeners {
+		if (listner.Protocol == gatewayv1beta1.UDPProtocolType) && (listner.Port == gatewayv1beta1.PortNumber(*udprouteSpec.Port)) {
+			return nil
+		}
+	}
+	return fmt.Errorf("No matching Gateway listener found for defined Parentref")
+}
+
+//+kubebuilder:rbac:groups=gateway.konghq.com,resources=udproutes,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=gateway.konghq.com,resources=udproutes/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=gateway.konghq.com,resources=udproutes/finalizers,verbs=update
+
+// UDProuteReconciler reconciles UDPRoute object
+func (r *UDPRouteReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	log := log.FromContext(ctx)
+
+	//Retrieve udp route object.
+	udproute := new(gatewayv1alpha2.UDPRoute)
+	if err := r.Get(ctx, req.NamespacedName, udproute); err != nil {
+		if errors.IsNotFound(err) {
+			log.Info("object enqueued no longer exists, skipping")
+			return ctrl.Result{}, nil
+		}
+		log.Info("Error retrieving udp route", "Err : ", err)
+		return ctrl.Result{}, err
+	}
+
+	//Use the retreive objects its parent ref to look for the gateway.
+	for _, parentRef := range udproute.Spec.ParentRefs {
+
+		//Build Gateway object to retreive
+		gw := new(gatewayv1beta1.Gateway)
+
+		ns := udproute.Namespace
+		if parentRef.Namespace != nil {
+			ns = string(*parentRef.Namespace)
+		}
+
+		//Get Gateway for UDP Route
+		if err := r.Get(ctx, types.NamespacedName{Name: string(parentRef.Name), Namespace: ns}, gw); err != nil {
+			log.Info("Unable to get parent ref gateway for", "udpRoute", parentRef.Name, "Namespace", ns, "err", err)
+			return ctrl.Result{}, err
+		}
+
+		//Check if referred gateway has the atleast one listner with properties defined from UDPRoute parentref.
+		if err := r.verifyListener(ctx, gw, parentRef); err != nil {
+			log.Info("No matching listener found for referred gateway", "GatewayName", parentRef.Name, "GatewayPort", parentRef.Port)
+			return ctrl.Result{}, err
```

Comment:
So by default, errors will trigger a re-queue. Since this condition (where we don't have a valid listener to attach to) is not a resolvable condition without end-user interference it would be better to stop and update the status of the `UDPRoute`. That would however be better to resolve as a separate smaller PR, so I've created a follow up for this: https://github.com/Kong/blixt/issues/39

Reply:
Will need help on this one, resolving here and will start discussion on #39  thread.